File: 658-find-k-closest-elements/find-k-closest-elements.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> findClosestElements(int[] arr, int k, int x) {
        int l = 0;
        int r = arr.length;

        while( l < r) {
            int m = l + (( r - l) / 2);

            if( m + k < arr.length && (x - arr[m]) > (arr[m + k] - x) ) {
                l = m + 1;
            }else {
                r = m;
            }
        }

        List<Integer> ans = new ArrayList<>();
        for(int i = l; i < l + k; i++) {
            ans.add(arr[i]);
        }

        return ans;
    }
}

File: 658-find-k-closest-elements/find-k-closest-elements.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @param {number} x
 * @return {number[]}
 */
var findClosestElements = function(arr, k, x) {

    let i = 0;
    let j = arr.length - 1;

    while(i < j) {
        let m = i + Math.floor(( j - i) / 2);
        if( x - arr[m] >  arr[m + k] - x ) i = m + 1;
        else j = m;
    }

    console.log(i, j);
    let ans = [];
    for(let m = i; m < i + k; m++) {
        ans.push(arr[m]);
    }

    return ans;
};